### PR TITLE
samples: Bluetooth: Broadcast audio source USB headset on nRF52 Series

### PR DESCRIPTION
**Pull request by cvinayak**
Broadcast audio source USB headset on nRF52 Series with USB support.

### CODE REVIEW

**Comment by larsgk** [comment]
This is awesome 😎 I would like to try it with an nRF52840 dongle too (if it works, you can transform any modern phone to be a broadcaster ;))

**Comment by larsgk** [comment]
@cvinayak  What is missing to move this PR out of draft?

**Comment by cvinayak** [comment]
> @cvinayak What is missing to move this PR out of draft?

i will need to test on a nRF52840 dongle, remove the nrf52833, have only overlay for nrf52840dongle (which should be sufficient), and update documentation. Hope to do these by end of this week, once In get a dongle to test.

**Comment by cvinayak** [comment]
> > > @cvinayak What is missing to move this PR out of draft?
> > 
> > 
> > i will need to test on a nRF52840 dongle, remove the nrf52833, have only overlay for nrf52840dongle (which should be sufficient), and update documentation. Hope to do these by end of this week, once I get a dongle to test.
> 
> also, for encrypted broadcast, I think only 1-2 (newest) nRF52x SoCs + nRF5340 work. For unencrypted, the dongle will be perfect. Do you have a dongle to test with?

Yes, only newer nRF5 SoCs will truely support encrypted Broadcast Audio, but any nRF52x and nRF53x series will work for cleartext broadcast audio.

I have now tested on nrf52840dongle_nrf52840  boards, please build your self and revert back if any issues...

**Comment by larsgk** [comment]
@cvinayak thanks for testing on the dongle - how was the quality?

Btw .. there is a PR with a bump of liblc3 https://github.com/zephyrproject-rtos/zephyr/pull/66017

Would it be possible for you to do a quick smoke test on the DK with the new version?

**Comment by cvinayak** [comment]
> @cvinayak thanks for testing on the dongle - how was the quality?
> 
> Btw .. there is a PR with a bump of liblc3 #66017
> 
> Would it be possible for you to do a quick smoke test on the DK with the new version?

Did a quick test, no regressions.

**Review by larsgk — DISMISSED** [review]

**Comment by larsgk** [comment]
> > @cvinayak thanks for testing on the dongle - how was the quality?
> > Btw .. there is a PR with a bump of liblc3 #66017
> > Would it be possible for you to do a quick smoke test on the DK with the new version?
> 
> Did a quick test, no regressions.

and I just flashed a dongle.  Clear audio reception in the other end... this... is awesome!!! Thanks @cvinayak

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CONFIG_MAIN_STACK_SIZE=4096
+
+# Use USB Audio as input
+CONFIG_USE_USB_AUDIO_INPUT=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr Broadcast Source"
+
+# Two streams in one subgroup (stereo)
+CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT=1
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf:8]: We could consider making this the default instead

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CONFIG_MAIN_STACK_SIZE=4096
+
+# Use USB Audio as input
+CONFIG_USE_USB_AUDIO_INPUT=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr Broadcast Source"
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf:5]: We could consider making this the default

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.overlay [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	hs_0: hs_0 {
+		compatible = "usb-audio-hs";
+		mic-feature-mute;
+		mic-channel-l;
+		mic-channel-r;
+
+		hp-feature-mute;
+		hp-channel-l;
+		hp-channel-r;
+	};
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.overlay:14]: These exists in the `.dtsi` for the nRF 5340 ADK;

Why is it different for this board?

File: samples/bluetooth/broadcast_audio_source/prj.conf [hunk 4]
```diff
@@ -13,3 +13,10 @@ CONFIG_BT_ISO_TX_BUF_COUNT=4
 CONFIG_BT_ISO_TX_MTU=60
 
 CONFIG_BT_DEVICE_NAME="Broadcast Audio Source"
+
+# program stack and CPU usage
+CONFIG_THREAD_ANALYZER=y
+CONFIG_THREAD_ANALYZER_USE_PRINTK=y
+CONFIG_THREAD_ANALYZER_AUTO=y
+CONFIG_THREAD_ANALYZER_AUTO_INTERVAL=5
+CONFIG_THREAD_NAME=y
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/prj.conf:22]: This should not be enabled by default in the sample IMO

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.overlay (hunk 3 above)
Comment (larsgk) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.overlay:14]: My guess is that the Audio DK is considered to use USB for audio more than the standard DK.

Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.overlay:14]: only CDC-ACM included by default in nRF SoC's with USB support, hence need an explicit overlay in this sample.

File: samples/bluetooth/broadcast_audio_source/prj.conf (hunk 4 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/prj.conf]: Removed

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf (hunk 2 above)
Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf:5]: Not all SoCs support USB, lets keep the USB Kconfig in board configs for now.

**Comment by cvinayak** [comment]
Updated to take advantage of per-transmissions.

File: samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf (hunk 2 above)
Comment (Thalley) [reply samples/bluetooth/broadcast_audio_source/boards/nrf52833dk_nrf52833.conf:5]: Fair

File: samples/bluetooth/broadcast_audio_source/prj.conf [hunk 5]
```diff
@@ -7,9 +7,10 @@ CONFIG_BT_BAP_BROADCAST_SOURCE=y
 
 CONFIG_BT_ISO_MAX_CHAN=2
 CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT=2
-CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT=2
-CONFIG_BT_ISO_TX_BUF_COUNT=4
+# Two streams in one subgroup (stereo)
+CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT=1
 
+CONFIG_BT_ISO_TX_BUF_COUNT=6
```
Comment (Thalley) [inline samples/bluetooth/broadcast_audio_source/prj.conf:13]: This effectively gives us 3 bufs per stream; wasn't 2 enough?

Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/prj.conf:13]: 2 buffers will be held per ISO interval and hence 1 additional for the next ISO interval.

Comment (cvinayak) [reply samples/bluetooth/broadcast_audio_source/prj.conf:13]: Tx buffers live/allocated across ISO Interval plus the delays; an SDU's PDU Tx buffer can remain allocated until all the required ISO intervals are exhausted.

In the case of 1 pre-transmission count and pre-transmission offset of 1, the SDU's PDU Tx buffer remains allocated for 2 adjacent ISO interval plus delay and a latency upto one ISO interval. Hence, to have new SDUs ready for subsequent ISO interval, additional n free PDU Tx buffers need be available for the application to be able to enqueue in time for the transmission.

**Review by larsgk — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**